test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -383,23 +379,15 @@ func vjpNonDiffResult2(x: Float) -> (Float, Int) {
 struct VJPStruct {
   let p: Float
 
-  @differentiable(vjp: storedPropVJP)
-  let storedImmutableOk: Float
```

Comment:
For type checking tests like this, I'd actually suggest turning them into computed properties. These tests are pretty important.